### DISCUSSION TITLE
onDisconnect Supabase Client event

### DISCUSSION THREAD

**Discussion by billypap1** [supabase/supabase#6061] [Questions] [answered]
Hello, I'm migrating my Firebase Realtime Database project to Supabase. 
I have created a table called "users" and I want to update the state of user  (online-offline). 
Each time the user logs in to my app, I update the table and set the column "state" to online. 
Is there any way to listen when Supabase Client disconnects in order to set the user's state to offline? 

In firebase for example I can use ->  userStatusDatabaseRef.onDisconnect().set(isOfflineForDatabase)

**Comment by GaryAustin1** [comment, marked as answer]
Don't know if this will help with your case if/when it is implemented:
https://github.com/supabase/supabase/discussions/5544#discussioncomment-2218497
